great_expectations/checkpoint/checkpoint.py

```diff
@@ -252,6 +262,21 @@ def _sort_actions(self) -> List[CheckpointAction]:
 
         return priority_actions + secondary_actions
 
+    def is_added(self) -> tuple[bool, list[ResourceNotAddedError]]:
+        errs: list[ResourceNotAddedError] = []
+
+        validations_added: bool = True
+        for validation_definition in self.validation_definitions:
+            validation_added, validation_errs = validation_definition.is_added()
+            errs.extend(validation_errs)
+            validations_added = validation_added and validations_added
+
+        self_added = self.id is not None
+        if not self_added:
+            errs.append(CheckpointNotAddedError(name=self.name))
+
+        return (validations_added and self_added, errs)
```

Comment:
I am fine with returning this as a tuple. Other alternatives include defining a class or dataclass.

Reply:
I think I'm going to put it in a dataclass just so the signature has some context - what's a good name 🤔

Reply:
Just keeping it as a tuple for simplicity's sake